sqlmesh/utils/__init__.py

```diff
@@ -236,3 +237,22 @@ def env_vars(environ: dict[str, str]) -> t.Iterator[None]:
     finally:
         os.environ.clear()
         os.environ.update(old_environ)
+
+
+def merge_dicts(*args: t.Dict) -> t.Dict:
+    """
+    Merges dicts. Just does key collision replacement
+    """
+
+    def merge(a: t.Dict, b: t.Dict) -> t.Dict:
+        for b_key, b_value in b.items():
+            a_value = a.get(b_key)
+            if b_key in a and isinstance(a_value, dict) and isinstance(b_value, dict):
+                merge(a_value, b_value)
+            elif isinstance(b_value, dict):
+                a[b_key] = copy.deepcopy(b_value)
```

Comment:
do you need the deepcopy here?

Reply:
Yes or else we end up with mutations on the input dicts.